build/commands/lib/start.js

```diff
@@ -84,6 +84,15 @@ const start = (passthroughArgs, buildConfig = config.defaultBuildConfig, options
     killSignal: 'SIGTERM'
   }
 
+  if (process.platform === 'darwin' && !config.isReleaseBuild()) {
+    // Disable 'accept incoming network connections' and 'keychain access'
+    // dialogs in MacOS. See //docs/mac_build_instructions.md for details.
+    braveArgs.push(
+      '--use-mock-keychain',
+      '--disable-features=DialMediaRouteProvider'
```

Comment:
this will override `--disable-features` passed by the user.

I think this can work if you add it before `braveArgs = braveArgs.concat(passthroughArgs)` line. Pls test this.

Reply:
added a check for `--disable-features`